examples/fluids/include/petsc_ops.h

```diff
@@ -11,6 +11,10 @@
 #include <ceed.h>
 #include <petscdm.h>
 
+#if PETSC_VERSION_GE(3, 21, 0)
+#define DMProjectCoordinates(a, b) DMSetCoordinateDisc(a, b, PETSC_TRUE)
+#endif
+
```

Comment:
```suggestion
#if PETSC_VERSION_LT(3, 21, 0)
#define DMSetCoordinateDisc(a, b, c) DMProjectCoordinates(a, b)
#endif

```
We can also do this, if we want to update the body of the code to match PETSc main.

Reply:
I tend to like this better-- a step forward that preserves backward compatibility with earlier PETSc versions. 

Note ProjectCoordinates is also used in stats.

Reply:
Perhaps this will also make it through CI  (which the original one did not) which is puzzling since that CI is with 3.19. so I think this should have done NOTHING.

Reply:
Anything that isn't a release commit reads as v∞.∞.∞

Reply:
So I tested the most recent commit  against an "older" PETSc 
```
grep VERSION_ $PETSC_DIR/configure.log
          PETSC_VERSION_RELEASE of 1 indicates the code is from a release branch or a branch created from a release branch.
          #define PETSC_VERSION_RELEASE    0
          #define PETSC_VERSION_MAJOR      3
          #define PETSC_VERSION_MINOR      19
          #define PETSC_VERSION_SUBMINOR   5
          #define PETSC_VERSION_DATE       "unknown"
          #define PETSC_VERSION_GIT        "unknown"
          #define PETSC_VERSION_DATE_GIT   "unknown"
          #define PETSC_VERSION_EQ(MAJOR,MINOR,SUBMINOR) \
          #define PETSC_VERSION_ PETSC_VERSION_EQ
          #define PETSC_VERSION_LT(MAJOR,MINOR,SUBMINOR)          \
          #define PETSC_VERSION_LE(MAJOR,MINOR,SUBMINOR) \
          #define PETSC_VERSION_GT(MAJOR,MINOR,SUBMINOR) \
          #define PETSC_VERSION_GE(MAJOR,MINOR,SUBMINOR) \
            Defined "VERSION_GIT" to ""v3.19.5-1159-g9fdbc8c8285""
            Defined "VERSION_DATE_GIT" to ""2023-09-30 13:10:58 -0600""
            Defined "VERSION_BRANCH_GIT" to ""kjansen/DMLoadCGNS_PlexSetBC""
        
 ```

and it compiles fine which means it is seeing and doing the #DEFINE but CI is still choking on it?

What am I missing?

Reply:
I guess that is what @jeremylt is saying 2 up from here, that this switch only work for releases and thus this is not going to work for us?

Reply:
I may have found something that works in this latest commit.

Reply:
Seems to work now.

Reply:
Oh, I see this thread now. There is no intent that old non-release PETSc snapshots will work. One should either run a release or run main. The ad-hoc inequality has other failure modes. I think your PETSc branch should be rebased on `main`, in which case the version check will also work.